packages/mgt-chat/src/utils/mentions.tsx

```diff
@@ -0,0 +1,36 @@
+import React from 'react';
+import { PersonCardInteraction } from '@microsoft/mgt-components';
+import { MgtTemplateProps, Person } from '@microsoft/mgt-react';
+import { ChatMessageMention, User } from '@microsoft/microsoft-graph-types';
+import { GraphChatClient } from 'src/statefulClient/StatefulGraphChatClient';
+import { Mention } from '@azure/communication-react';
+
+export const renderMGTMention = (chatState: GraphChatClient) => {
+  return (mention: Mention, defaultRenderer: (mention: Mention) => JSX.Element): JSX.Element => {
+    let render: JSX.Element = defaultRenderer(mention);
+
+    const mentions = chatState?.mentions ?? [];
+    // TODO: Array.flat() is a new api, update?
+    // eslint-disable-next-line @typescript-eslint/no-unnecessary-type-assertion
+    const flatMentions = mentions?.flat() as ChatMessageMention[];
+    // eslint-disable-next-line @typescript-eslint/non-nullable-type-assertion-style
+    const teamsMention = flatMentions.find(
+      m => m.id?.toString() === mention?.id && m.mentionText === mention?.displayText
+    ) as ChatMessageMention;
```

Comment:
I'm a little suspect of this line.

if mentions is nullish then flatMentions could also be nullish.
find can return undefined.
can mention really be nullish? if so, we should refactor this whole flow to check that condition early and not do any processing of the chatState.mentions.

```suggestion
    const teamsMention?: ChatMessageMention = flatMentions?.find(
      m => m.id?.toString() === mention?.id && m.mentionText === mention?.displayText
    );
```

Reply:
Mentions can be nullish. Also, their type is a `NullableOption<ChatMention>`. I'm updating this suggestion to take care of the scenario. Thanks.